turbo/engineapi/engine_server.go

```diff
@@ -570,20 +574,23 @@ func (s *EngineServer) forkchoiceUpdated(ctx context.Context, forkchoiceState *e
 	}, nil
 }
 
-func (s *EngineServer) getPayloadBodiesByHash(ctx context.Context, request []libcommon.Hash, _ clparams.StateVersion) ([]*engine_types.ExecutionPayloadBodyV1, error) {
+func (s *EngineServer) getPayloadBodiesByHash(ctx context.Context, request []libcommon.Hash, version clparams.StateVersion) ([]*engine_types.ExecutionPayloadBody, error) {
+	if len(request) > 1024 {
```

Comment:
Is it in the spec somewhere?

Reply:
Yes, see point 3 at https://github.com/ethereum/execution-apis/blob/main/src/engine/shanghai.md#specification-3
It does say "The call MUST return `-38004: Too large request` error if the requested range is too large" with a subjective definition of "too large"